### ISSUE TITLE
I am using version 0.7.1 with dockerd enabled. When I start VS code from WSL (my code is in the filesystem of an Ubuntu 20.04 distribution) and then try to start my devcontainer, this fails with the following error message:

### ISSUE THREAD

**Issue by jandubois** [rancher-sandbox/rancher-desktop#1186]
I am using version 0.7.1 with dockerd enabled. When I start VS code from WSL (my code is in the filesystem of an Ubuntu 20.04 distribution) and then try to start my devcontainer, this fails with the following error message:

`docker: Error response from daemon: invalid mount config for type "bind": invalid mount path: '1ddff10d-b7f5-4248-87dc-c16fa415c643' mount path must be absolute.`

The command that fails is the following:

`docker run --sig-proxy=false -a STDOUT -a STDERR --mount type=bind,source=/home/ckihm/workspace/sbss-example-service,target=/workspaces/sbss-example-service --mount type=volume,src=vscode,dst=/vscode --mount type=bind,src=/var/run/docker.sock,dst=/var/run/docker-host.sock -l vsch.local.folder=\\wsl$\Ubuntu-DevEnv\home\ckihm\workspace\sbss-example-service -l vsch.quality=stable -l vsch.remote.devPort=0 --init --entrypoint /bin/sh vsc-sbss-example-service-877b6f700ecd25f2f1f2e8038c325ff9-features-uid -c echo Container started`

However, the paths are correct. If I try in the shell to resolve the used paths then this works without problems.

I am quite sure that this is the only problem that currently prevents the use of VS Code devcontainer with rancher desktop. I have also tried to prefix the paths with \\wsl$\Ubuntu-DevEnv\ and prober quotet the backslashes, but without luck. 

If you have any hint which might solve this issue, I would appreciate it.

_Originally posted by @ckihm in https://github.com/rancher-sandbox/rancher-desktop/issues/800#issuecomment-1000391333_

**Comment by jandubois** [comment]
@ckihm I've created a new issue from your comment because the original issue was closed, but since you mention RD 0.7.1, I assume your problem still persists.

**Comment by n0v1** [comment]
I'm getting a similar error with Rancher Desktop 0.7.1 in dockerd mode when trying to run a container that has a file secret mounted with docker-compose 1.29.2 from within WSL2:

`docker-compose.yml`
```yaml
version: '3.4'

services:
  my-service:
    image: "hello-world"
    secrets:
      - my_secrets.json

secrets:
  my_secrets.json:
    file: ./my-secrets.json
```
`my-secrets.json`
```json
{
  "foo": "bar"
}
```

```bash
# docker-compose up --force-recreate
Recreating 0c696893af91_docker_my-service_1 ... error

ERROR: for 0c696893af91_docker_my-service_1  Cannot create container for service my-service: invalid mount config for type "bind": invalid mount path: '14d3186d-747d-4eb9-be97-81c055b99fb3' mount path must be absolute
```

The same error occurs when using an absolute path to `my-secrets.json` in the compose file. Both variants work fine when using Docker Desktop.

**Comment by jandubois** [comment]
The fix has already shipped in 1.0.0-beta.1. Please verify that your issue has been solved!

**Comment by n0v1** [comment]
Bind mounting volumes works for me in 1.0.0-beta.1. Also I don't get the `invalid mount path` error anymore when using a file secret. But `/run/secrets/my_secrets.json` is an empty directory instead of a file with content from `my-secrets.json`.

```yaml
version: '3.4'

services:
  my-service:
    image: "alpine"
    secrets:
      - my_secrets.json
    command: "ls -l /run/secrets/"

secrets:
  my_secrets.json:
    file: ./my-secrets.json
```

```bash
# docker-compose up --force-recreate
Recreating docker_my-service_1 ... done
Attaching to docker_my-service_1
my-service_1  | total 0
my-service_1  | drwx------    2 root     root            40 Jan 17 10:16 my_secrets.json
```
